### PR TITLE
Update book_learn extraction in lang/extract_json_strings.py

### LINKED ISSUE #42901: Translation extraction script creates incorrect entries
# Describe the bug
I see two incorrect entries foк translation in Transofex master file, namely just `o` and `g` entries that shouldn't be there as they don't represent actual line to be translated.

# Steps To Reproduce
Check transifex `master` file and find lines 25665-25966
See `o` and `g` entries

# Expected behavior
Python script not extracting invalid entries for translation

# Versions and configuration
N/A

# Additional context
I was adviced to create this issue after some discission on Discord. 
@anothersimulacrum pointed that line
`"mag_glam": { "skill_level": 5 }, "cookbook_liverforkids":`
is what causing the problem.

### PR DESCRIPTION
**Pull request by AMurkin**
#### Summary

SUMMARY: None

#### Purpose of change

Fixes #42901

#### Describe the solution

Parse both array and dictionary book_learn values.
Updated all book_learn in json to dictionary format.

#### Describe alternatives you've considered

Leaving a `Good First Issue` for another contributor.

#### Testing

No 'g' and 'o' strings extracted, and new string extracted:
```diff
 _('openair, covered work space')
-_('g')
-_('o')
+_('Buttery Duck Bites')
 _('Candied Onions and Giblets')
```

### CODE REVIEW

File: lang/extract_json_strings.py [hunk 1]
```diff
@@ -583,9 +583,14 @@ def extract_monster_attack(item):
...
-            if len(arr) >= 3 and len(arr[2]) > 0:
-                writestr(outfile, arr[2])
+        if type(item["book_learn"]) is dict:
+            for (k, v) in item["book_learn"].items():
+                if type(v) is dict and "recipe_name" in v:
+                    writestr(outfile, v["recipe_name"])
+        else:
+            for arr in item["book_learn"]:
+                if len(arr) >= 3 and len(arr[2]) > 0:
+                    writestr(outfile, arr[2])
```
Comment (anothersimulacrum) [inline lang/extract_json_strings.py:593]: The game cannot read things in this format anymore, you should just remove the code for it.

Comment (AMurkin) [reply lang/extract_json_strings.py:593]: Doesn't it extract that string?
```json
  {
    "type": "recipe",
    "result": "liver_onion",
    "result_mult": 2,
    "category": "CC_FOOD",
    "subcategory": "CSC_FOOD_MEAT",
    "skill_used": "cooking",
    "difficulty": 2,
    "time": "30 m",
    "batch_time_factors": [ 83, 5 ],
    "book_learn": [ [ "scots_cookbook", 2 ], [ "cookbook", 0 ], [ "cookbook_liverforkids", 1, "Candied Onions and Giblets" ] ],
    "qualities": [ { "id": "CUT", "level": 1 }, { "id": "COOK", "level": 2 } ],
    "tools": [ [ [ "surface_heat", 50, "LIST" ] ] ],
    "components": [ [ [ "liver", 8 ] ], [ [ "onion", 1 ], [ "irradiated_onion", 1 ] ] ]
  }
```
Game logic:
https://github.com/CleverRaven/Cataclysm-DDA/blob/478ced98fd31acbaef7dd6903d31572a562bbebd/src/recipe.cpp#L228-L237

Comment (anothersimulacrum) [reply lang/extract_json_strings.py:593]: No, that array extraction code does not extract a string, it only extracts the `0` and `1` elements, the book and skill.

**Comment by AMurkin** [comment]
I should also update more json to not lose these lines:
```
-_("Stuff THE MAN doesn't want you to know")
-_('Sticky Rice Hedgerows')
-_('Low-Carb Sashimi Donburi')
-_('Royal Penny Tea')
-_('Sir Tenderloin the Toothsome')
-_('Shagbark Nut Ambrosia')
-_('Natural Beet Sugar')
-_('Leprechaun Sausage')
-_('Spaghetti Luchetto')
```

**Pull request merged**